File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        if(nums.size() <= 1) return 0; 
        
        int jump = 0; 
        int currEnd = 0; 
        int maxJump = 0; 
        for(int i = 0; i < nums.size()-1; i++){
            maxJump = max(maxJump, i + nums[i]); 
            if(i == currEnd){
                jump++; 
                currEnd = maxJump; 
            }
        }
        return jump; 
    }
};

File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        if(nums.length == 1) return 0; 
        int answer = 0; 
        int currJump = 0; 
        int currMax = 0; 

        for(int i = 0; i < nums.length; i++){
            currMax = Math.max(currMax, i + nums[i]); 
            if(i >= currJump){
                answer++;
                currJump = currMax; 
                if(currJump >= nums.length-1) return answer;  
            }
        }
        return -1; 
    }
}